api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/EriDSDCohortQueries.java

```diff
@@ -770,58 +770,48 @@ public CohortDefinition getN2() {
     CohortDefinition rapidFlow = getPatientsWithStartOrContinueOnRapidFlow();
 
     String mappings = "onOrBefore=${endDate},location=${location}";
-    //  cd.addSearch("TxCurr", EptsReportUtils.map(txCurr, mappings));
-    //  cd.addSearch("scheduledN2", EptsReportUtils.map(patientsScheduled, mappings));
+    cd.addSearch("TxCurr", EptsReportUtils.map(txCurr, mappings));
+    cd.addSearch("scheduledN2", EptsReportUtils.map(patientsScheduled, mappings));
 
-    String rapidFlowMappings =
-        "onOrAfter=${startDate},onOrBefore=${endDate},locationList=${location}";
     cd.addSearch("rapidFlow", EptsReportUtils.map(rapidFlow, mappings));
-    /* cd.addSearch(
-        "completed", EptsReportUtils.map(getPatientsWhoCompletedRapidFlow(), mappings));
+    cd.addSearch("completed", EptsReportUtils.map(getPatientsWhoCompletedRapidFlow(), mappings));
 
     cd.addSearch(
         "nonPregnantNonBreastFeedingNonTb",
         EptsReportUtils.map(
             getPregnantAndBreastfeedingAndOnTBTreatment(),
             "endDate=${endDate},location=${location}"));
-    /*cd.setCompositionString(
-        "TxCurr AND (scheduledN2 OR rapidFlow) NOT (completed OR nonPregnantNonBreastFeedingNonTb)");*/
-    cd.setCompositionString("rapidFlow");
+    cd.setCompositionString(
+        "TxCurr AND (scheduledN2 OR rapidFlow) NOT (completed OR nonPregnantNonBreastFeedingNonTb)");
 
     return cd;
   }
 
-  private CohortDefinition getPatientsWhoCompletedRapidFlow() {
-    CodedObsCohortDefinition cd = new CodedObsCohortDefinition();
+  public CohortDefinition getPatientsWhoCompletedRapidFlow() {
+    SqlCohortDefinition cd = new SqlCohortDefinition();
+
     cd.addParameter(new Parameter("onOrBefore", "Before Date", Date.class));
-    cd.addParameter(new Parameter("onOrAfter", "After Date", Date.class));
     cd.addParameter(new Parameter("location", "Location", Location.class));
-    CommonQueries.getLastCodedObsBeforeDate(
-        hivMetadata.getAdultoSeguimentoEncounterType().getEncounterTypeId(),
-        hivMetadata.getRapidFlow().getConceptId(),
-        Arrays.asList(
-            hivMetadata.getStartDrugsConcept().getConceptId(),
-            hivMetadata.getContinueRegimenConcept().getConceptId()));
+    cd.setQuery(
+        CommonQueries.getLastCodedObsBeforeDate(
+            Arrays.asList(hivMetadata.getAdultoSeguimentoEncounterType().getEncounterTypeId()),
+            hivMetadata.getRapidFlow().getConceptId(),
+            Arrays.asList(hivMetadata.getCompletedConcept().getConceptId())));
     return cd;
   }
 
-  private CohortDefinition getPatientsWithStartOrContinueOnRapidFlow() {
-    CodedObsCohortDefinition cd = new CodedObsCohortDefinition();
+  public CohortDefinition getPatientsWithStartOrContinueOnRapidFlow() {
+    SqlCohortDefinition cd = new SqlCohortDefinition();
```

Comment:
give this definition a name, it enables easier debugging, log follow up and where errors emerge, you get to know where everything fell  guided by the named objects

Reply:
@ningosi sorry, i didn't catch very well that part, can u give an example of the definition name so i can update ?

Reply:
Sorry, I meant if you give all the definition names would be great. Something like
cd.setName("som string here) I would name it as **Patients who started or continuing on rapid flow**

Reply:
@ningosi Yes. let me do that. in 2mins, it will help a lot giving a name on this.